Tests/SentryTests/Integrations/Performance/FramesTracking/SentryFramesTrackerTests.swift

```diff
@@ -710,6 +710,156 @@ class SentryFramesTrackerTests: XCTestCase {
+        
+        // Pause the tracker
+        sut.pause()
+        
+        // Verify it's paused
+        XCTAssertFalse(sut.isRunning)
+        
+        // Unpause and verify the previous frame timestamp is reset
+        sut.unpause()
+        XCTAssertTrue(sut.isRunning)
+        
+        // The next frame should be treated as the first frame (previousFrameTimestamp == SentryPreviousFrameInitialValue)
+        // This means it won't be classified as slow/frozen even if there was a long pause
+        fixture.displayLinkWrapper.call()
+        
+        // Should not detect any slow or frozen frames after unpausing
+        try assert(slow: 0, frozen: 0, total: 2)
+    }
+    
+    func testUnpause_AfterLongPause_DoesNotDetectHang() throws {
+        let sut = fixture.sut
+        sut.start()
+        
+        // Simulate some normal frames
+        fixture.displayLinkWrapper.call()
+        fixture.displayLinkWrapper.normalFrame()
+        
+        // Pause the tracker
+        sut.pause()
+        
+        // Simulate a very long pause (longer than frozen frame threshold)
+        fixture.dateProvider.advance(by: 5.0) // Much longer than 0.7s frozen threshold
+        
+        // Unpause the tracker
+        sut.unpause()
+        
+        // The next frame should not be classified as frozen even though there was a long pause
+        fixture.displayLinkWrapper.call()
+        
+        // Should not detect any slow or frozen frames
+        try assert(slow: 0, frozen: 0, total: 1)
+    }
+    
+    func testUnpause_WhenAlreadyRunning_DoesNotResetTimestamp() throws {
+        let sut = fixture.sut
+        sut.start()
+        
+        // Simulate some frames
+        fixture.displayLinkWrapper.call()
+        fixture.displayLinkWrapper.normalFrame()
+        
+        // Try to unpause when already running
+        sut.unpause()
+        
+        // Should still be running
+        XCTAssertTrue(sut.isRunning)
+        
+        // Continue with normal frames
+        fixture.displayLinkWrapper.normalFrame()
+        
+        // Should have normal frame counting behavior
+        try assert(slow: 0, frozen: 0, total: 2)
+    }
+    
+    func testUnpause_AfterBackgroundForegroundTransition_ResetsTimestamp() throws {
+        let sut = fixture.sut
+        sut.start()
+        
+        // Simulate some frames
+        fixture.displayLinkWrapper.call()
+        fixture.displayLinkWrapper.normalFrame()
+        
+        // Simulate app going to background
+        fixture.notificationCenter.post(Notification(name: CrossPlatformApplication.willResignActiveNotification))
+        XCTAssertFalse(sut.isRunning)
+        
+        // Simulate a long time in background
+        fixture.dateProvider.advance(by: 10.0)
+        
+        // Simulate app coming to foreground
+        fixture.notificationCenter.post(Notification(name: CrossPlatformApplication.didBecomeActiveNotification))
+        XCTAssertTrue(sut.isRunning)
```

Comment:
`l`: Yes this is how we indirectly call pause and unpause. We can merge this test case with `testUnpause_AfterLongPause_DoesNotDetectHang`. We only need one. They are basically testing the same thing.